### PR TITLE
numpy 1.21.1 testing

### LINKED ISSUE #2613: PythonMedianImageFilterTest fails with numpy 1.21.0
### Description

`PythonMedianImageFilterTest` fails after numpy is upgraded from 1.20.3 to the recently released 1.21.0.

### Steps to Reproduce

Enable `ITK_WRAP_PYTHON` and install `numpy==1.21.0`.

### Expected behavior

This test should not fail.

### Actual behavior

Fails with:

```none
2916: Traceback (most recent call last):
2916:   File "/home/matt/src/ITK/Modules/Filtering/Smoothing/wrapping/test/MedianImageFilterTest.py", line 56, in <module>
2916:     type_hints = get_type_hints(itk.median_image_filter, globalns= { 'itk': itk })
2916:   File "/home/matt/bin/miniconda/lib/python3.8/typing.py", line 1266, in get_type_hints
2916:     value = _eval_type(value, globalns, localns)
2916:   File "/home/matt/bin/miniconda/lib/python3.8/typing.py", line 272, in _eval_type
2916:     ev_args = tuple(_eval_type(a, globalns, localns) for a in t.__args__)
2916:   File "/home/matt/bin/miniconda/lib/python3.8/typing.py", line 272, in <genexpr>
2916:     ev_args = tuple(_eval_type(a, globalns, localns) for a in t.__args__)
2916:   File "/home/matt/bin/miniconda/lib/python3.8/typing.py", line 272, in _eval_type
2916:     ev_args = tuple(_eval_type(a, globalns, localns) for a in t.__args__)
2916:   File "/home/matt/bin/miniconda/lib/python3.8/typing.py", line 272, in <genexpr>
2916:     ev_args = tuple(_eval_type(a, globalns, localns) for a in t.__args__)
2916:   File "/home/matt/bin/miniconda/lib/python3.8/typing.py", line 270, in _eval_type
2916:     return t._evaluate(globalns, localns)
2916:   File "/home/matt/bin/miniconda/lib/python3.8/typing.py", line 520, in _evaluate
2916:     eval(self.__forward_code__, globalns, localns),
2916:   File "<string>", line 1, in <module>
2916: NameError: name 'dtype' is not defined
2916: itkTestDriver: Process exited with return value: 1```

### Reproducibility

Always.

### Versions

Current `master` or `release`, 5.2.0.

### Environment

Locally using Python 3.8.

### Additional Information

This seems to be related to the use of `numpy.typing.ArrayLike`.

### PR DESCRIPTION
**Pull request by thewtex**
- COMP: Revert "COMP: Use numpy==1.20.3 for CI testing". Closes #2613.

### CODE REVIEW

File: Testing/ContinuousIntegration/AzurePipelinesMacOSPython.yml [hunk 1]
```diff
@@ -26,7 +26,8 @@ jobs:
 
     - bash: |
         set -x
-        sudo pip3 install ninja numpy==1.20.3 typing-extensions
+        sudo pip3 install ninja numpy
```
Comment (dzenanz) [inline Testing/ContinuousIntegration/AzurePipelinesMacOSPython.yml:29]: unintentionally repeated line?

Comment (thewtex) [reply Testing/ContinuousIntegration/AzurePipelinesMacOSPython.yml:29]: Good catch, yes

**Review by jhlegarreta — APPROVED** [review]
:+1: Fixes #2613, right?

**Comment by Leengit** [comment]
It looks like [tensorflow requirements](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/pip_package/setup.py#L78-L81) still include numpy 1.19.2.  Building ITK with a newer numpy by default should be okay, but it would be good if ITK also remains backward compatible for a while.

**Comment by thewtex** [comment]
> It looks like tensorflow requirements still include numpy 1.19.2. Building ITK with a newer numpy by default should be okay, but it would be good if ITK also remains backward compatible for a while.

Yes, we are not changing the backwards compatibility with this change.

**Pull request merged**